src/pages/Lectures.tsx

```diff
@@ -0,0 +1,198 @@
+                    groupedLectures[groupedLectures.length - 1].lectures.push(lect)
+                } else {
+                    //else it starts a new group
+                    groupedLectures.push({date: lect.date, lectures: [lect]} as LectureGrouped)
+                }
+            }
+
+        })
+        return groupedLectures
+    }
+    /**
+     * this function formats the date so there gets back a clear date and a clear day so everyone knows which day it is
+     * */
+    const formatDate = function (date: Date): string {
+        let dateString: string = "";
+        if (date.getDate() < 10) {
+            dateString = "0" + date.getDate() + ". "
+        } else {
+            dateString = date.getDate() + ". "
+        }
+        switch (date.getMonth()) {
+            case 0:
+                dateString += "Jan";
+                break;
+            case 1:
+                dateString += "Feb";
+                break;
+            case 2:
+                dateString += "Mär";
+                break;
+            case 3:
+                dateString += "Apr";
+                break;
+            case 4:
+                dateString += "Mai";
+                break;
+            case 5:
+                dateString += "Jun";
+                break;
+            case 6:
+                dateString += "Jul";
+                break;
+            case 7:
+                dateString += "Aug";
+                break;
+            case 8:
+                dateString += "Sep";
+                break;
+            case 9:
+                dateString += "Okt";
+                break;
+            case 10:
+                dateString += "Nov";
+                break;
+            case 11:
+                dateString += "Dez";
+                break;
+        }
+        dateString += " " + date.getFullYear() + " - ";
+        switch (date.getDay()) {
+            case 1:
+                dateString += "Montag";
+                break;
+            case 2:
+                dateString += "Dienstag";
+                break;
+            case 3:
+                dateString += "Mittwoch";
+                break;
+            case 4:
+                dateString += "Donnerstag";
+                break;
+            case 5:
+                dateString += "Freitag";
+                break;
+            case 6:
+                dateString += "Samstag";
+                break;
+            case 7:
+                dateString += "Sonntag";
+                break;
+        }
+        return dateString
+    }
+
+    //states for refresh button and lectures
+    const [lectures, setLectures] = useState([] as LectureGrouped[]);
+    const [refreshing, setRefresh] = useState(false);
+
+
+    useEffect(() => {
+        getData('lectures').then((lectureList) => {
+            if (lectureList) {
+                const lect: Lecture[] = lectureList.map((lecture: Lecture) => {
+                    const lec: Lecture = {
+                        id: lecture["id"],
```

Comment:
Hier kann man auch Object spreading verwenden:
```
const lec: Lecture = {
…lecture,

date: new Date(lecture["date"]),
startTime: new Date(lecture["startTime"]),
endTime: new Date(lecture["endTime"]),
}

```

Reply:
Auch in https://github.com/stuv-mosbach/StuVReactNativeFrontend/pull/3/files#diff-cf80019a768bf6052ef824adf44049cdf4e7641ddb2a1f2552df9aa74946153eR21-R30.
Eventuell den doppelten Code in eine gemeinsam genutze Funktion verschieben.